torchvision/io/image.py

```diff
@@ -37,7 +37,7 @@ def decode_png(input):
     Returns:
         output (Tensor[image_width, image_height, 3])
     """
-    if not isinstance(input, torch.Tensor) or input.numel() == 0 or input.ndim != 1:
+    if not isinstance(input, torch.Tensor) or input.numel() == 0 or input.dim() != 1:
```

Comment:
any reason to change `ndim` to `dim()`? `ndim` is a new attribute of tensor, which follows numpy more closely.

Reply:
`mypy` doesn't recognize the `ndim` attribute:

```python
import torch

tensor = torch.empty(1, 2, 3)

print(tensor.dim())
print(tensor.ndim)
```

```
test_ndim.py:6: error: "Tensor" has no attribute "ndim"  [attr-defined]
    print(tensor.ndim)
          ^
Found 1 error in 1 file (checked 1 source file)
```

Either we use `dim()` or I add an ignore. If we later use [`warn_unused_ignores`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-warn-unused-ignores) we can simply remove the ignore as soon as this is fixed upstream.

Reply:
Can you open an issue in PyTorch mentioning the `ndim` doesn't have mypy support?

Reply:
Already reported and fixed https://github.com/pytorch/pytorch/issues/42908

Reply:
Do you want to go with `dim()` or `ndim` and an ignore?

Reply:
If it's been fixed, can we keep `ndim`?